### ISSUE TITLE
Add a NgForm.revalidate() method

### ISSUE THREAD

**Issue by tony75** [angular/angular.dart#1598]
Hi,

It would be great to have a revalidate() method that reset the errorState of each form childControl for this use case : 

 - A field in the form has some ng- constraints (required, minlength, etc.). Ng Constraints errors are automatically added and removed from the field errorState.
- When the form is submitted, an http request is done and sends data to the server.
- If there are some errors, for instance an incorrect password, the server returns the map of the incorrect fields with their error. Then, the errors are added on the fields using the addErrorState method.

The problem is that when the user re-submit the form, it is still flagged invalid, because the errors added manually from the server response are still here. To solve such a problem, when the user re-submit the form we actually need to loop on each field to remove from the errorStates all the custom errors .

A revalidate() method could do that automatically by clearing the errorStates maps, and revalidating the form fields.

**Comment by matsko** [comment]
@tony75 does the `form.reset()` method do the trick? #1637 fixes a pending issue with `reset()`. Once the reset fix is merged in then perhaps adding the following to ngControl would do the trick?

```dart
void validate() {
_controls.forEach((control) => control.validate());
}
```

Is that what you're after?

**Comment by matsko** [comment]
By the looks of it `form.reset()` does a reset of all models which in turn causes a call to `model.validate()`. I don't see much of a reason to have `form.validate()` since all the input-related DOM operations are handled (which in turn cause something to validate).

**Comment by tony75** [comment]
@matsko This is not exaclty the use case I was after because the reset will overwrite all the models to set the original values, but I want to keep the modified values.

For instance, if a user fill a form, but type a wrong password, I add an error in the password model errorState (all the errors comming from the server are automatically added to their errorState corresponding model) which causes the form to be invalid. 
When the user resubmit the form, I want to remove these errors added from the server response, but without overwriting the modifications done by the user, so that he does not have to fill again the entire form.

**Comment by matsko** [comment]
OK. So from what I understand. We need to have the following behaviour:

```dart
formModel.clearErrors();
// or passwordModel.clearErrors();
// or passwordModel.errorStates.clear(); ... this works now, but it doesn't remove any CSS classes or flags

// then
formModel.validate();
```

**Comment by tony75** [comment]
That's it :)

**Comment by matsko** [comment]
Here's the PR: https://github.com/angular/angular.dart/pull/1639

**Comment by tony75** [comment]
@matsko It looks good, thanks a lot for this PR

**Comment by rkirov** [comment]
merged #1639
